File: 728.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> ans = new ArrayList<>();
        for(int i = left; i <= right;i ++) {
            if(helper(i)) ans.add(i);
        }
        return ans;
    }

    public boolean helper(int n) {
        int cur = n;
        while(cur != 0) {
            if(cur%10 == 0 || n%(cur%10) != 0) return false;
            else cur /= 10;
        }
        return true;
    }
}

File: 728.go
func selfDividingNumbers(left int, right int) []int {
	ans := []int{}
	for i := left; i <= right; i++ {
		if helper(i) {
			ans = append(ans, i)
		}
	}
	return ans
}

func helper(n int) bool {
	cur := n
	for cur != 0 {
		if cur%10 == 0 || n%(cur%10) != 0 {
			return false
		} else {
			cur /= 10
		}
	}
	return true
}